src/main/java/radiounit/NokiaLteRadioUnitReceiver.java

```diff
@@ -13,60 +15,185 @@
-		// TODO Auto-generated method stub
-
+		System.out.println(
+				"[NokiaLteRadioUnitReceiver] modifyCarrierNokiaLte: " + carrierId + ", " + frequencyBand.getBand());
+		
+		int index = getCarrierIndexById(carrierId);
+		
+		// check that carrier exists
+		if (index == CARRIER_NOT_FOUND_INDEX) {
+			System.err.println("Carrier with ID [" + "] does not exist!");
+			return;
+		}
+		
+		Carrier existingCarrier = getCarrierById(carrierId);
+		
+		removeCarrierInternal(index);
+		existingCarrier.setFrequencyBand(frequencyBand);
+		addCarrierInternal(existingCarrier);
 	}
 
 	public void removeCarrierNokiaLte(Integer carrierId) {
-		// TODO Auto-generated method stub
-
+		System.out.println("[NokiaLteRadioUnitReceiver] removeCarrierNokiaLte: " + carrierId);
+		
+		int index = getCarrierIndexById(carrierId);
+		
+		// check that carrier exists
+		if (index == CARRIER_NOT_FOUND_INDEX) {
+			System.err.println("Carrier with ID [" + "] does not exist!");
+			return;
+		}
+		
+		removeCarrierInternal(index);
 	}
 
 	public void selfDiagnosticsNokiaLte() {
-		// TODO Auto-generated method stub
-
+		System.out.println("[NokiaLteRadioUnitReceiver] selfDiagnosticsNokiaLte");
 	}
 
 	public void removeAllCarriersNokiaLte() {
-		// TODO Auto-generated method stub
-
+		System.out.println("[NokiaLteRadioUnitReceiver] removeAllCarriersNokiaLte");
+		
+		for (int i = 0; i < getAllCarriersInternal().size(); ++i) {
+			removeCarrierInternal(i);
+		}
 	}
 
 	public List<Carrier> getCarriers() {
-		return carriers;
+		
+		List<Carrier> carrierList = getAllCarriersInternal();
+		
+		for (Carrier c : carrierList) {
+			System.out.println("Carrier: " + c);
+		}
+		
+		return carrierList;
+	}
+	
+	private Carrier getCarrierById(int id)
+	{
+		int index = getCarrierIndexById(id);
+
+		if (index == CARRIER_NOT_FOUND_INDEX)
+		{
+			return null;
+		}
+
+		return getAllCarriersInternal().get(index);
 	}
 
+	private int getCarrierIndexById(int id)
+	{
+		for (int i = 0; i < getAllCarriersInternal().size(); ++i)
+		{
+			Carrier c = getAllCarriersInternal().get(i);
+
+			if (c.getCarrierId() == id)
+			{
+				return i;
+			}
+		}
+
+		return CARRIER_NOT_FOUND_INDEX;
+	}
+
+	/**
+	 * Private, internal method with exclusive "getter" access to the carriers list.
+	 * @param idx Index of the carrier to be retrieved
+	 * @return Carrier at the specified index
+	 */
+	private synchronized List<Carrier> getAllCarriersInternal()
+	{
+		return this.carriers;
+	}
+
+	/**
+	 * Private, internal method with exclusive "setter" access to the carriers list.
+	 * @param carrier Carrier to add to the list
+	 */
+	private synchronized void addCarrierInternal(Carrier carrier)
+	{
+		this.carriers.add(carrier);
+	}
+
+	/**
+	 * Private, internal method with exclusive removal access to the carriers list.
+	 * @param index Index of the carrier to be removed
+	 */
+	private synchronized void removeCarrierInternal(int index)
```

Comment:
Not a necessary change at all but just wanted to add it as a helpful tip in the future. Instead of the conditional here, you could try/catch an IndexOutOfBounds exception. It saves a comparison by the end of it, which can be nice at times.

Reply:
Ah, so just having a try/catch instead of checking to see if it is within the index? I hadn't thought of that, and would be curious to try it out. Thanks for the suggestion.